zkp/circuits/lib/encrypt.circom

```diff
@@ -24,16 +25,63 @@ template SymmetricEncrypt(length) {
   signal input plainText[length];  // private
   signal input key[2];  // private
   signal input nonce;  // public
-  signal output cipherText[length];
 
-  component hashers[length];
-  for (var i = 0; i < length; i++) {
-    hashers[i] = Poseidon(4);
-    hashers[i].inputs[0] <== key[0];
-    hashers[i].inputs[1] <== key[1];
-    hashers[i].inputs[2] <== nonce;
-    hashers[i].inputs[3] <== i;
+  var two128 = 2 ** 128;
 
-    cipherText[i] <== hashers[i].out + plainText[i];
+  // nonce must be < 2^128
+  component lt = LessThan(252);
+  lt.in[0] <== nonce;
+  lt.in[1] <== two128;
+  lt.out === 1;
+
+  // the number of plain text messages must be multiple of 3
+  // pad the array with zeros if necessary.
+  // e.g. if length == 4, l == 6
+  var l = length;
+  while (l % 3 != 0) {
+    l += 1;
+  }
```

Comment:
```suggestion
  var l = length;
  if (l % 3 != 0) {
      l += (3 - (l % 3));
  }
```

I tend to avoid `while` loop in other programming languages, thus the suggestion. But I don't know whether this will improve/reduce performance in the circom world.

Reply:
I understand the coding style. and thanks for suggesting the optimization